documentation/docs/extensibility.md

```diff
@@ -1,86 +1,199 @@
+
+For the version identifier, please see the section _How to stay up-to-date_ in this document.
+
+### Multiple projects
+
+If you have multiple projects that share a similar architecture, it might be desirable to share one modified pipeline amongst them.
+Similar to what you can do in an individual `Jenkinsfile`, you can copy the pipeline to your own shared library and modify it.
+
+To do this, create a new git repository in your preferred git hosting service.
+It must be compliant to [how Jenkins shared libraries are built](https://jenkins.io/doc/book/pipeline/shared-libraries/).
+In a nutshell, this means you need a `vars` directory inside which you can place a copy of your preferred pipeline.
+
+A minimal example of such a library could have this directory structure:
+
+```
+./vars/myCustomPipeline.groovy
+./README.md
+```
+
+Where `myCustomPipeline.groovy` contains the modified pipeline code of the [general purpose pipeline](https://github.com/SAP/jenkins-library/blob/master/vars/piperPipeline.groovy) or [SAP Cloud SDK Pipeline](https://github.com/SAP/cloud-s4-sdk-pipeline-lib/blob/master/vars/cloudSdkPipeline.groovy).
+
+!!! note
+    Your custom pipeline _must_ be named differently from the other pipelines provided by project "Piper", because Jenkins requires names across multiple libraries to be unique.
 
-## 2. Central Custom Template
+This library must be placed in a git repository which is available for Jenkins and must be configured in Jenkins [as documented here](https://jenkins.io/doc/book/pipeline/shared-libraries/#using-libraries).
 
-If you have multiple projects where you want to use a custom template, you could implement this similarly to [piperPipeline](https://github.com/SAP/jenkins-library/blob/master/vars/piperPipeline.groovy).
+This screenshot shows an example of the configuration in Jenkins.
+Note the name (1) must be the same you use in your `Jenkinsfile`.
 
-!!! note "How to not get decoupled"
-    Typically, providing a custom template decouples you from centrally provided updates to your template including the stages.<br />
-    Where applicable, you can re-use the stage implementations. This means, you will call e.g. `piperPipelineStageBuild()` as you can see in [piperPipeline](https://github.com/SAP/jenkins-library/blob/master/vars/piperPipeline.groovy).
+![Library Setup](images/customPipelineLib.png "Library Setup")
 
-    Using this approach you can at least benefit from innovations done in individual project "Piper" stages.
+The `Jenkinsfile` of your individual projects would look similar to this:
 
-!!! note "When to go with a custom template"
-    If the configuration possibilities are not sufficient for you and if _1. Stage Exits_ is not applicable.
+```groovy
+@Library(['piper-lib-os@vINSERT_VERSION_HERE','my-own-pipeline@vINSERT_VERSION_HERE']) _
 
-## 3. Custom Jenkinsfile
+myCustomPipeline script: this
```

Comment:
You could also load a custom pipeline (Jenkinsfile) that is located in another repo.

Reply:
Not sure if I understand your comment. This part describes how to share a custom pipeline (as part of a jenkins lib) amongst multiple projects. What would you change?

Reply:
Maybe I didn't position the comment very good. My comment refers to the whole sub-chapter "__Multiple projects__". For multiple projects you suggest defining a pipeline in a custom lib, which IMO is a relatively complicated approach. When instead it would be possible to have a modified pipeline (as suggested in the "__Single project__" approach) in another repo and load that from multiple projects. 

IMO the lib approach is to add functionality that you don't have otherwise, _not_ to define pipelines.

Reply:
I yield then.